src/platform/packages/shared/kbn-esql-ast/src/types.ts

```diff
@@ -130,9 +130,9 @@ export interface ESQLAstCompletionCommand extends ESQLCommand<'completion'> {
 }
 
 export interface ESQLAstRerankCommand extends ESQLCommand<'rerank'> {
-  query: ESQLLiteral;
+  query: ESQLLiteral | ESQLUnknownItem;
   fields: ESQLAstField[];
-  inferenceId: ESQLIdentifierOrParam;
```

Comment:
The grammar and this PR https://github.com/elastic/elasticsearch/issues/124337 example suggest to use a generic map with one or more key -> values and not just inference_id after the WITH token.